R/api.R

```diff
@@ -191,22 +205,37 @@ handleAPIfailure <- function(code, response) {
         return(crGET(response$url))
     }
     msg <- http_status(response)$message
+    msg2 <- NULL
     if (code == 404) {
         # Add the URL that was "not found" (there isn't going to be any
         # useful response content message)
         msg2 <- response$url
     } else {
-        msg2 <- try(content(response)$message, silent = TRUE)
+        err_content <- try(content(response), silent = TRUE)
```

Comment:
Are both of the checks needed in the top-level if statement? I'd think `is.list` is enough ...

Do we lose anything with a more radical simplification such as:

```
if (is.list(err_content)){
    msg2 <- coalesce(err_content$message, err_content$description)
}
```

... adding coalesce somewhere:

```
coalesce <- function(...){
    for (i in list(...)) if (!is.null(i)) return(i)
}
```

Reply:
Cool, I'll take another look. Partly I'm trying to be as defensive as possible because an error here would mean an unintelligible error to the user

Reply:
Adapted some, but really want to be conservative here because we're by definition working with a response we don't fully understand (because we sent an error) and it makes super confusing behavior for the user if we error in our intrepration